### PR TITLE
feat(api): auth api implementation

### PR DESCRIPTION
**Pull request by nikhilverma360**
- Added a new endpoint for authenticating users via POST request. Users provide a token and username, which are verified against the provided JWT token.
- Added functions for interacting with the Supabase database. These functions handle operations such as setting and getting user information, checking user existence, and retrieving user connections.
- Added initial database schema for storing user and leaderboard information.
- Added a trigger and function to automatically add new users to the leaderboard upon registration.

### CODE REVIEW

File: scripts/qr-generator.js [hunk 1]
```diff
@@ -22,10 +22,10 @@ if (!SECRET_KEY || !BASE_URL) {
 const amount = process.argv[2] ? parseInt(process.argv[2]) : 1;
 
 for (let i = 1; i <= amount; i++) {
-  const userId = uuidv4();
-  const payload = { userId };
+  const user_id = uuidv4();
+  const payload = { user_id };
   const token = jwt.sign(payload, SECRET_KEY);
-  const qr_png = qr.image(BASE_URL+"?" + token, {
+  const qr_png = qr.image(BASE_URL+"/welcome?token=" + token, {
```
Comment (nikhilverma360) [inline scripts/qr-generator.js]: I have changed the url to `/welcome?token=sometoken` according to figma naming
or should we go for `/auth?token=sometoken`

Comment (alcercu) [reply scripts/qr-generator.js]: I think I prefer `/auth?token=sometoken`, then once the user is authenticated, we can redirect it to `/welcome` if needed.

File: src/lib/supabase/queries.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+export const getUser = async (user_id: UUID) => {
+  const { data, error } = await supabase
+    .from("users")
+    .select()
+    .eq("user_id", user_id)
+    .single();
+  return { data, error };
+};
+
+export const isUserExist = async (user_id: UUID) => {
```
Comment (alcercu) [inline src/lib/supabase/queries.ts]: I would change the name of this function to `checkUserExists` or

File: src/lib/supabase/queries.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+};
+
+export const getUser = async (user_id: UUID) => {
+  const { data, error } = await supabase
+    .from("users")
+    .select()
+    .eq("user_id", user_id)
+    .single();
+  return { data, error };
+};
```
Comment (alcercu) [inline src/lib/supabase/queries.ts:24]: I believe there is a way to export supabase typing for typescript, that way we can indicate the types for these functions.

**Review by alcercu — CHANGES_REQUESTED** [review]

File: src/lib/supabase/queries.ts (hunk 3 above)
Comment (nikhilverma360) [reply src/lib/supabase/queries.ts:24]: added 🫡

File: scripts/qr-generator.js (hunk 1 above)
Comment (nikhilverma360) [reply scripts/qr-generator.js]: done🫡

**Review by alcercu — APPROVED** [review]
lgtm

**Pull request merged**